src/deployment.rs

```diff
@@ -79,6 +85,118 @@ pub struct Artifact {
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+
+        Ok(Self {
+            js_key: None,
+            wasm_file_name: None,
+            serve_path
+        })
+    }
+
+    fn create_serve_path_for(&mut self, name: &str, ext: &::std::ffi::OsStr) -> String {
+        let serve_path = match self.serve_path {
+            Some(ref path) => path,
+            None => return name.to_string()
+        };
+        let with_path = PathBuf::from( serve_path ).join( name ).to_string_lossy().to_string();
+        if ext == "js" {
+            self.js_key = Some( with_path.clone() );
+        } else if ext == "wasm" {
+            self.wasm_file_name = Some( name.to_string() );
+        }
+        with_path
+    }
+
+    fn insert_serve_path_to_js(&self, routes: &mut Vec<Route>) {
+        let serve_path = match self.serve_path {
+            Some(ref path) => path,
+            None => return
+        };
+
+        let js_key = if let Some(ref js_key) = self.js_key {
+            js_key
+        } else {
+            return;
+        };
+
+        let wasm_u8 = if let Some(ref wasm_file_name) = self.wasm_file_name {
+            wasm_file_name.as_bytes()
+        } else {
+            return;
+        };
+
+        let _buf = PathBuf::from( serve_path ); // because of temporary value does not live long enough
+        let _lossy = _buf.to_string_lossy(); // because of temporary value does not live long enough
+        let serve_path_u8 = _lossy.as_bytes();
+
+        // Search backward for value in contents
+        fn search_start_index(contents: &[u8], value: &[u8], start_at: usize) -> Option<usize> {
+            for index in (0..start_at).rev() {
+                if contents[index..].starts_with(value) {
+                    return Some(index);
+                }
+            }
+            None
+        }
+
+        let js_route = routes.iter_mut().find(|r| r.key == *js_key).unwrap();
+        match js_route.kind {
+            RouteKind::Blob(ref mut contents) => {
+                contents.reserve( serve_path_u8.len() * 2 );
+
+                if let Some(start_index) 
+                    = search_start_index( &contents[..], wasm_u8, contents.len()-wasm_u8.len() )
+                {
+                    let _removed: Vec<u8> = 
+                        contents.splice( start_index..start_index, serve_path_u8.iter().cloned() ).collect();
+                    assert_eq!(_removed.len(), 0);
+
+                    if let Some(start_index)
+                        = search_start_index( &contents[..], wasm_u8, start_index )
+                    {
+                        let _removed: Vec<u8> =
+                            contents.splice( start_index..start_index, serve_path_u8.iter().cloned() ).collect();
+                        assert_eq!(_removed.len(), 0);
+                    }
+
+                }
+            }
+            _ => unreachable!()
+        }
```

Comment:
This looks a little too complicated; we shouldn't (mostly) need these functions, I think.

It would be simpler to just to do something like this:

```rust
fn rebase_url( deploy_path: &str, url: &str ) -> String {
    // ...
}
```

and then:

```diff
         for path in result.artifacts() {
             let (is_js, key) = match path.extension() {
-                Some( ext ) if ext == "js" => (true, js_name.clone()),
+                Some( ext ) if ext == "js" => (true, rebase_url( &deploy_path, &js_name )),
-                Some( ext ) if ext == "wasm" => (false, path.file_name().unwrap().to_string_lossy().into_owned()),
+                Some( ext ) if ext == "wasm" => (false, rebase_url( &deploy_path, path.file_name().unwrap().to_string_lossy() )),
                 _ => continue
             };
```

Reply:
As I mentioned this in my first reply above. We need to know the `actual name`  of the `.wasm` file (its name only, not include the `serve_path`, because what we want to insert the `serve_path` into `js`, before its `actual name`).